src/main/java/com/eprosima/fastdds/idl/templates/TypeObjectTestingTestSource.stg

```diff
@@ -0,0 +1,1437 @@
+        {
+            EXPECT_TRUE((key_type_ids.type_identifier1() == *type_ids.type_identifier1().map_ldefn().key_identifier() &&
+                    key_type_ids.type_identifier2() == *type_ids.type_identifier2().map_ldefn().key_identifier()) ||
+                    (key_type_ids.type_identifier1() == *type_ids.type_identifier2().map_ldefn().key_identifier() &&
+                    key_type_ids.type_identifier2() == *type_ids.type_identifier1().map_ldefn().key_identifier()));
+        }
+        else
+        {
+            EXPECT_EQ(key_type_ids.type_identifier1(), *type_ids.type_identifier1().map_ldefn().key_identifier());
+            EXPECT_EQ(*type_ids.type_identifier1().map_ldefn().key_identifier(), *type_ids.type_identifier2().map_ldefn().key_identifier());
+        }
+    }
+    EXPECT_EQ($map_name(map=map)$_expected_flags, type_ids.type_identifier1().map_ldefn().header().element_flags());
+    EXPECT_EQ($if(map.unbound)$0$else$$map.evaluatedMaxsize$$endif$, type_ids.type_identifier1().map_ldefn().bound());
+    EXPECT_EQ(key_$map_name(map=map)$_expected_flags, type_ids.type_identifier1().map_ldefn().key_flags());
+}
+>>
+
+test_string(string) ::= <<
+TEST(TypeObjectTests, TestStringTypeObject_$string_name(string=string)$)
+{
+    register_$ctx.filename$_type_objects();
+
+    $check_string_type(string=string)$
+}
+>>
+
+check_string_type(string) ::= <<
+ReturnCode_t ret_code;
+TypeIdentifierPair type_ids;
+$get_type_identifier_registry(typename=string_name(string), result="type_ids")$
+EXPECT_EQ(TK_NONE, type_ids.type_identifier2()._d());
+EXPECT_TRUE(TI_STRING8_SMALL == type_ids.type_identifier1()._d() || TI_STRING8_LARGE == type_ids.type_identifier1()._d());
+if (type_ids.type_identifier1()._d() == TI_STRING8_SMALL)
+{
+    EXPECT_EQ($if(string.isBounded)$$string.evaluatedMaxsize$$else$0$endif$, type_ids.type_identifier1().string_sdefn().bound());
+}
+else
+{
+    EXPECT_EQ($if(string.isBounded)$$string.evaluatedMaxsize$$else$0$endif$, type_ids.type_identifier1().string_ldefn().bound());
+}
+>>
+
+test_wstring(wstring) ::= <<
+TEST(TypeObjectTests, TestWstringTypeObject_$wstring_name(wstring=wstring)$)
+{
+    register_$ctx.filename$_type_objects();
+
+    $check_wstring_type(wstring=wstring)$
+}
+>>
+
+check_wstring_type(wstring) ::= <<
+ReturnCode_t ret_code;
+TypeIdentifierPair type_ids;
+$get_type_identifier_registry(typename=wstring_name(wstring), result="type_ids")$
+EXPECT_EQ(TK_NONE, type_ids.type_identifier2()._d());
+EXPECT_TRUE(TI_STRING16_SMALL == type_ids.type_identifier1()._d() || TI_STRING16_LARGE == type_ids.type_identifier1()._d());
+if (type_ids.type_identifier1()._d() == TI_STRING16_SMALL)
+{
+    EXPECT_EQ($if(wstring.isBounded)$$wstring.evaluatedMaxsize$$else$0$endif$, type_ids.type_identifier1().string_sdefn().bound());
+}
+else
+{
+    EXPECT_EQ($if(wstring.isBounded)$$wstring.evaluatedMaxsize$$else$0$endif$, type_ids.type_identifier1().string_ldefn().bound());
+}
+>>
+
+fwd_decl(ctx, parent, type) ::= <<>>
```

Comment:
Maybe its better to have at the top the mandatory StringTemplate rules and the auxiliary at the bottom.